Content.Tests/DMProject/Tests/Builtins/gradient.dm

```diff
@@ -0,0 +1,17 @@
+﻿// We do HSV gradients our own way
+
+/proc/RunTest()
+	var/list/gradlist = list(0, "#ff0000", 1, "#00ff00")
+	ASSERT(gradient("#ff0000", "#00ff00", 0.5) == "#7f7f00")
+	ASSERT(gradient(gradlist, 0.5) == "#7f7f00")
+	ASSERT(gradient(gradlist, 0) == "#ff0000")
+	ASSERT(gradient(gradlist, 1) == "#00ff00")
+	ASSERT(gradient(gradlist, "loop", 1.5) == "#7f7f00")
+	ASSERT(gradient(gradlist, "loop", 1.1) == "#e51900")
+	ASSERT(gradient(gradlist, "loop", 2) == "#ff0000")
+	ASSERT(gradient(gradlist, "loop", 1) == "#ff0000")
```

Comment:
These are not valid `gradient()` calls, they give `runtime error: missing index` in BYOND.

I think when the first argument is a list, the second argument _must_ be the index of the gradient and `"loop"` is not a valid index.

Reply:
I mean its fine if we allow a bit more shenanigans (although I'd love to remove it completely and be a list/index only)